dojo/models.py

```diff
@@ -438,14 +440,19 @@ def __unicode__(self):
         from urlparse import uses_netloc
 
         netloc = self.host
+        fqdn = self.fqdn
+        port = self.port
         scheme = self.protocol
         url = self.path if self.path else ''
         query = self.query
         fragment = self.fragment
 
+        if port:
+            netloc += ':%s' % port
+
         if netloc or (scheme and scheme in uses_netloc and url[:2] != '//'):
             if url and url[:1] != '/': url = '/' + url
-            if scheme:
+            if scheme and scheme in uses_netloc and url[:2] != '//':
```

Comment:
I've done some testing against local files on this and it seems to work but the logic here is complicated enough that I don't fully trust my own judgment here...so please test this :)